OpenSim/Simulation/Model/Model.cpp

```diff
@@ -975,14 +975,11 @@ void Model::extendConnectToModel(Model &model)
     // TODO: Get rid of the SimbodyEngine
     updSimbodyEngine().connectSimbodyEngineToModel(*this);
 
-    // Cache enabled controllers locally so that `computeControls` can
-    // iterate over the cache, rather than needing to call
-    // `getComponentList` (expensive) on each iteration.
-    this->_enabledControllers.clear();
+    // Now that the model is fully connected, cache controllers
+    // locally to reduce the runtime cost of Model::computeControls()
+    this->_controllers.clear();
```

Comment:
Your `_enabledControllers` was correct. All that needed to change was that `Controller:isEnabled()` should not be checking the flag `getAllControllersEnabled` on `Model`. It should have remained local. Model has a chance to override all computing of controls at compute time. Sorry if that was not clear on my previous comments.

Reply:
Yeah, but changing `Controllers::isEnabled` is something I'm deliberately avoided in that patch to get things running again, because I'll need to verify that changing the enablement logic will not break anything. The enablement logic effectively means that, if the root is enabled, all controllers are enabled..

Changing controllers to only use their local `isEnabled` logic is correct, but I avoided it in this first patch because I don't know how the defaulting logic etc. of `Controller`'s enabled member works if (e.g.) it's constructed without `enabled` explicitly being set to `true` in the XML or whatever.

I'm currently digging through the various layers of object setup abstraction because I (as always) don't know the initialization defaults etc. of the various classes, and the members are all defined via macros that set property lookups in a `SimTK::Array_<AbstractProperty*>` bla bla. As always, writing as I go along: if only to make concrete what I'm learning along the way. Imho, stepping through OpenSim + SimTK's abstractions is a fairly hilarious insight.

Reply:
The current `Controller:isEnabled()` implementation is a bug. Please do fix it. I missed it before. The  fix is much more clear. Any individual actuator can be disabled (effectively excluded) from the System (prior to System creation) which is in agreement with our notion of properties and has no dependency on its owner. The `Model::_allControllersEnabled` flag is used to exclude all Controllers at compute time. It is used by a few algorithms.  I am quite certain (99%) that anything that breaks is a related bug that should be fixed (but more likely nothing will break).

Reply:
Related, we can enable Travis CI on this repo so we can verify if there is any fallout.